ios/HackerNews/Updates/AutoUpdatemanager.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  AutoUpdatemanager.swift
+//  HackerNews
+//
+//  Created by Itay Brenner on 31/1/25.
+//
+
+#if ADHOC
+import Foundation
+import ETDistribution
+
+struct AutoUpdateManager {
+  @MainActor static func checkForUpdates() {
```

Comment:
Can this use async/await with a checked continuation? Would love to use async everywhere possible. 😈